### PR TITLE
Destroy a Session (Aka logout)

### PR DESCRIPTION
**Pull request by Roberto-Cardenas**
### What
Added the functionality for destroying a session (aka logging out of the system). Modified necessary views, helpers, routes, and controller actions. Also added the code for checking if a user is considered active when logging in and testing for that feature.

### How
Added a hamburger menu that when clicked displays a side menu with a link for logging out. Modified the controller action responsible for destroying a session as well as fixing the routing. Wrote some simple behavior tests.

@armandofox

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ def login(user)
     session[:user_id] = user.id
   end
 
+  def logout
+    session[:user_id] = nil
```
Comment (armandofox) [inline app/helpers/sessions_helper.rb]: a slightly more idiomatic way to do this is `session.delete(:user_id)` which removes the key entirely from the hash.

File: app/views/announcements/index.html.erb [hunk 2]
```diff
@@ -2,18 +2,21 @@
 <div class = "sticky-top">
   <!-- General Nav bar -->
   <nav class="navbar navbar-expand-lg navbar-light bg-dark p-0">
-    <div class="container-fluid">
-      <a class="navbar-brand" href="#">
-        <%= image_tag "bsfc_logo.png", {:alt => "Berkeley Student Food Collective Logo", :width => 40, :height => 40} %>
+    <div class = "container-fluid">
```
Comment (armandofox) [inline app/views/announcements/index.html.erb]: spaces around `=` signs are an antipattern for HTML.  it looks like these files were correct before but this error got introduced. pls clean up.

File: config/routes.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@
   root 'sessions#new'
   get 'login' => 'sessions#new'
   post 'login' => 'sessions#create'
-  delete 'logout' => 'sessions#destroy'
+  get 'logout' => 'sessions#destroy'
```
Comment (armandofox) [inline config/routes.rb]: unless there's a really good reason, "destructive" or actions with side effects (like logout) should be accessed via a `post` or some action other than `get`.  strongly suggest modifying the route and UI button accordingly.

File: features/session/logout.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      | name             | email                     |
+      | Roberto Cardenas | robertocardenas@bsfc.coop |
+      | Andrew Mo        | andrewmo@bsfc.coop        |
+
+  When I start on the login page
+  When I fill in "session_email" with "robertocardenas@bsfc.coop"
+  And I press "Login"
+
+  Scenario: User wants to logout from the Announcements page
+    Given I should see "Announcements"
```
Comment (armandofox) [inline features/session/logout.feature]: this is minor but for readability, you should be more careful about the use of Given, When, and Then.  Given sets up preconditions, When simluates a user action, Then asserts postconditions.  in my view, for readability, line 13 above should be And, and `Then I should  see "Announcements"` seems like it should be part of the background steps (to confirm the login actually happened).  also that will help with the scenario of lines 16-19 currently not making sense: "Given i should see 'announcmenets' and I follow 'log out'" doesn't make sense as a story.  let me know if this comment makes sense, if not i can elaborate.

**Review by armandofox — CHANGES_REQUESTED** [review]

**Comment by Roberto-Cardenas** [comment]
@armandofox 
I think we are ready for re-review

**Review by armandofox — APPROVED** [review]
please confirm before i merge this:  i do not see tests for logout, have these been implemented?

**Comment by Roberto-Cardenas** [comment]
Theres only behavior testing for logout its under features/session/logout.feature

**Pull request merged**